index.js

```diff
@@ -291,17 +291,27 @@
       'https://github.com/sanctuary-js/sanctuary-type-classes/tree/v' + version
         + '#' + _name,
       dependencies,
-      function(x) {
-        return (
-          staticMethodNames.every (function(_name) {
-            return x != null &&
-                   getStaticMethod (_name) (x.constructor) != null;
-          }) &&
-          prototypeMethodNames.every (function(_name) {
-            return getPrototypeMethod (_name) (x) != null;
-          })
-        );
-      }
+      (function() {
+        var $seen = [];
+        return function(x) {
+//        if ($seen.includes (x)) return true;
```

Comment:
I don't think this code path can be exercised without the changes in the following commit. I think `npm test` should pass for every new commit on `master`, so I commented out this line and uncommented it in the following commit.